OpenRA.Mods.Common/TraitsInterfaces.cs

```diff
@@ -208,75 +208,75 @@ public interface INotifyHarvestAction
 		void MovementCancelled(Actor self);
 	}
 
-	public interface IDockClientInfo : ITraitInfoInterface { }
+	public interface ILinkClientInfo : ITraitInfoInterface { }
 
-	public interface IDockClient
+	public interface ILinkClient
 	{
-		BitSet<DockType> GetDockType { get; }
+		BitSet<LinkType> LinkType { get; }
 
 		/// <summary>When null, the client should act as if it can dock but never do.</summary>
-		DockClientManager DockClientManager { get; }
-		void OnDockStarted(Actor self, Actor hostActor, IDockHost host);
-		bool OnDockTick(Actor self, Actor hostActor, IDockHost dock);
-		void OnDockCompleted(Actor self, Actor hostActor, IDockHost host);
+		LinkClientManager LinkClientManager { get; }
+		void OnLinkCreated(Actor self, Actor hostActor, ILinkHost host);
+		bool OnLinkTick(Actor self, Actor hostActor, ILinkHost dock);
+		void OnLinkRemoved(Actor self, Actor hostActor, ILinkHost host);
 
 		/// <summary>Is this client allowed to dock.</summary>
 		/// <remarks>
-		/// Does not check if <see cref="Traits.DockClientManager"/> is enabled.
-		/// Function should only be called from within <see cref="IDockClient"/> or <see cref="Traits.DockClientManager"/>.
+		/// Does not check if <see cref="Traits.LinkClientManager"/> is enabled.
+		/// Function should only be called from within <see cref="ILinkClient"/> or <see cref="Traits.LinkClientManager"/>.
 		/// </remarks>
-		bool IsDockingPossible(BitSet<DockType> type, bool forceEnter = false);
+		bool IsLinkingPossible(BitSet<LinkType> type, bool forceEnter = false);
 
 		/// <summary>Is this client allowed to dock to <paramref name="host"/>.</summary>
 		/// <remarks>
-		/// Does not check if <see cref="Traits.DockClientManager"/> is enabled.
-		/// Function should only be called from within <see cref="IDockClient"/> or <see cref="Traits.DockClientManager"/>.
+		/// Does not check if <see cref="Traits.LinkClientManager"/> is enabled.
+		/// Function should only be called from within <see cref="ILinkClient"/> or <see cref="Traits.LinkClientManager"/>.
 		/// </remarks>
-		bool CanDockAt(Actor hostActor, IDockHost host, bool forceEnter = false, bool ignoreOccupancy = false);
+		bool CanLinkTo(Actor hostActor, ILinkHost host, bool forceEnter = false, bool ignoreOccupancy = false);
 	}
 
-	public interface IDockHostInfo : ITraitInfoInterface { }
+	public interface ILinkHostInfo : ITraitInfoInterface { }
 
-	public interface IDockHost
+	public interface ILinkHost
 	{
-		BitSet<DockType> GetDockType { get; }
+		BitSet<LinkType> GetLinkType { get; }
 
 		/// <summary>Use this function instead of ConditionalTrait.IsTraitDisabled.</summary>
 		bool IsEnabledAndInWorld { get; }
 		int ReservationCount { get; }
 		bool CanBeReserved { get; }
-		WPos DockPosition { get; }
-		int DockWait { get; }
-		WAngle DockAngle { get; }
+		WPos LinkPosition { get; }
```

Comment:
These really should move out of the link interface and into the docking implementation.

Reply:
It's used by `FindAndDeliverResources` to search resources closest to the refinery, by CarryableHarvester to order carryalls to, and `LinkExts` to search for the closest Link

Reply:
It's exposed for outside traits to interact with the linking process

Reply:
Again, having a single `LinkPosition` is not correct for all potential types of link host. The interface should be abstracting the actions, not the implementation details (e.g. units should MoveWithinRange of a proximity host and not Move to a specific point).

Reply:
Sure, but I don't know the best way to do it. This can be iterated on in the future

Reply:
Isn't the problem then that FindAndDeliverResources depends on Link and not Dock?